c_glib/test/test-stream-decoder.rb

```diff
@@ -79,8 +79,15 @@ def test_consume_bytes
   end
 
   def test_consume_buffer
+    # We need to keep data that aren't processed yet.
```

Comment:
Does it make any sense to proactively change `test_consume_bytes` too? Line 71 in this file.

Reply:
No. We don't need to change `test_consume_bytes`.
`arrow::ipc::StreamDecoder::Consume(const uint8_t* data, int64_t size)` copied the given data internally when they aren't enough size.
(`arrow::ipc::StreamDecoder::Consume(std::shared_ptr<Buffer> buffer)` refers them instead of copying.)

Reply:
Why doesn't the `Buffer` object keep the Ruby object alive? PyArrow wouldn't have this problem AFAIR.